installer/pyinstaller/build-linux.sh

```diff
@@ -12,7 +12,7 @@ if [ "$python_library_zip_filename" = "" ]; then
 fi
 
 if [ "$python_version" = "" ]; then
-    python_version="3.11.8";
+    python_version="3.11.10";
```

Comment:
Should we also bump the python versions in GH actions?

Reply:
Looks like we don't pin [them](https://github.com/aws/aws-sam-cli/blob/develop/.github/workflows/validate_pyinstaller.yml), so should be alright.

Reply:
I thought for GH actions, you can't specify the minor version, just the major version, hopefully it will just use the latest